### PR TITLE
[Merged by Bors] - feat: Product of injective functions on sets

### PR DESCRIPTION
**Pull request by YaelDillies**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Set/Function.lean:1868]: I find dot notation for InjOn doesn't really read well, but won't block on it

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Set/Function.lean
Comment (urkud) [reply Mathlib/Data/Set/Function.lean:1868]: Why not use `Prod.map` in the statements?

File: Mathlib/Data/Set/Function.lean
Comment (urkud) [inline Mathlib/Data/Set/Function.lean:1046]: Alternative proof: `by rw [← hf.image_eq, forall_mem_image]`

File: Mathlib/Data/Set/Function.lean
Comment (YaelDillies) [reply Mathlib/Data/Set/Function.lean:1868]: because it gets simplified to its definition as `Prod_map` (not a typo!!) is a simp lemma

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**
